nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -196,7 +196,7 @@ in
       };
 
       timeout = mkOption {
-        default = 5;
+        default = if (config.boot.loader.timeout != null) then config.boot.loader.timeout else -1;
```

Comment:
This is also why I thought deprecating isn't a good idea. It's all still valid to use, but it might have non-obvious consequences. Is it better to just deprecate `config.boot.loader.grub.timeout`?